src/compiler/comments.ts

```diff
@@ -68,38 +68,48 @@ namespace ts {
                 emitDetachedComments,
             };
 
-            function getLeadingComments(range: TextRange | Node, getAdditionalRange?: (range: Node) => Node) {
-                let comments = getLeadingCommentsOfPosition(range.pos);
-                if (getAdditionalRange) {
-                    let additionalRange = getAdditionalRange(<Node>range);
-                    while (additionalRange) {
-                        comments = concatenate(
-                            getLeadingCommentsOfPosition(additionalRange.pos),
-                            comments
-                        );
-
-                        additionalRange = getAdditionalRange(additionalRange);
+            function getLeadingComments(range: TextRange | Node, shouldSkipCommentsForNodeCallback?: (node: Node) => boolean) {
+                if (shouldSkipCommentsForNodeCallback && shouldSkipCommentsForNodeCallback(<Node>range)) {
+                    // If the node will not be emitted in JS, remove all the comments(normal, pinned and ///) associated with the node,

```

Comment:
Space after "comments", comma after "pinned"

You say you're going to omit pinned comments here, but you commented at https://github.com/Microsoft/TypeScript/pull/7783/files#diff-c0d0e8b1528663b6cff3bb893150cec3R271 that a `NotEmittedStatement` will have its pinned comments omitted.